_posts/2024-02-01-wildfly-glow.adoc

```diff
@@ -0,0 +1,504 @@
+
+To enable Bootable JAR packaging, set the `<bootable-jar>true</bootable-jar>` plugin option.
+
+Note: Details on how to produce WildFly Bootable JAR from the WildFly Maven Plugin will come in a following blog post.
+
+=== Galleon CLI vs WildFly Glow CLI
+
+==== Server with support for postgresql datasource
+
+===== Galleon CLI
+
+[source,bash]
+----
+galleon.sh install wildfly --layers=cloud-server,ejb,jsf --dir=server
+galleon.sh install org.wildfly:wildfly-datasources-galleon-pack:6.0.0.Final --layers=postgresql-datasource --dir=server
+----
+
+Note: you will then have to deploy your application into the provisioned server.
+
+===== WildFly Glow CLI
+
+[source,bash]
+----
+wildfly-glow.sh scan myapp.war --add-ons=postgresql --provision=SERVER
+----
+Note: the provisioned server contains the deployment that has been scanned.
+
+Note: by specifying `--provision=BOOTABLE_JAR` you can produce a WildFly bootable JAR. By specifying `--provision=DOCKER_IMAGE` you can produce a Docker image.
+
+Producing a Docker image for cloud execution example:
+
+[source,bash]
+----
+wildfly-glow.sh scan myapp.war --add-ons=postgresql --provision=DOCKER_IMAGE --cloud
+----
+
+===== Output example
+
+WildFly Glow CLI output what it has discovered, for example:
+
+[source,bash]
+----
+
+wildfly-glow.sh scan myapp.war 
+
+Wildfly Glow is scanning...
+context: bare-metal
+enabled profile: none
+galleon discovery
+- feature-packs
+   org.wildfly:wildfly-galleon-pack:30.0.1.Final
+- layers
+   ee-core-profile-server
+   jpa
+   ejb-lite
+   jaxrs
+   jsf
+   h2-driver
+
+Some suggestions have been found. You could enable suggestions with --suggest option (if using the WildFly Glow CLI) or <suggest>true</suggest> (if using the WildFly Maven Plugin).
+To provision the WildFly server for your deployment add the --provision=SERVER option to the scan command
+
+----
+
+== Understanding why a Galleon layer has been selected
+
+WildFly Glow is based on rules contained in the Wildfy Galleon layers. If a rule matches, the layer is included. The set of rules 
+contained in WildFly layers is documented in this http://docs.wildfly.org/wildfly-galleon-feature-packs/[documentation].
+
+When enabling verbose mode (`--verbose` option with WildFly Glow CLI, `<verbose>true</verbose>` option in the maven plugin), 
+WildFly Glow will output for each layer the rules that matched.
+
+An example of output:
+
+----
+...
+layers inclusion rules
+* ee-core-profile-server
+  - BASE_LAYER
+* ee-concurrency
+  - JAVA_TYPE: [jakarta.enterprise.concurrent.*]
+* undertow-https
+  - ADD_ON
+...
+----
+
+This output means:
+
+* `ee-core-profile-server` is a base layer (always included).
+* `ee-concurrency` layer is included because a Java class located in the java package `jakarta.enterprise.concurrent` has been found.
+* `undertow-https` is included because it is bound to an included add-on (in this case `ssl`).
```

Comment:
Maybe we should have the CLI command used to invoke this? Or there is a chance I'll refer to the previous 'wildfly-glow.sh scan myapp.war' and get confused since I don't see any add-ons specified there